portfolio/src/main/webapp/script.js

```diff
@@ -58,17 +58,17 @@ function displayComments(){
   
   fetch('/data').then(response => response.json()).then((commentHistory) => {
     const commentSection = document.getElementById("existing-comments");
-    if((typeof commentHistory === 'string')&&(commentHistory.localeCompare("No comments found.")==0)) {
+    if((typeof commentHistory === 'string') && (commentHistory.localeCompare("No comments found.")==0)) {
```

Comment:
Using a string type to indicate no comments makes this code a little awkward. Returning an empty list and doing a length check in this if would be simpler. In general you want each function to only ever return one type.